teps/0160-enhance-results-cli.md

```diff
@@ -0,0 +1,202 @@
+- Improve accessibility and usability over the existing CLI.
+- Enable users to retrieve and analyze PipelineRun and TaskRun results efficiently.
+- Ensure authentication mechanisms are secure and easy to configure.
+- Allow the CLI to function both as a standalone tool and as a plugin to tkn CLI.
+
+### Non-Goals
+
+This proposal is not intended to modify the archived PipelineRuns and TaskRuns
+
+### Use Cases
+
+- As a `Developer`, I want to `list` my past PipelineRuns and TaskRuns without writing custom API queries.
+- As a `Tekton admin`, I want to `describe` specific runs to debug failures.
+- As a `DevOps engineer`, I want to `fetch logs` of past runs.
+- As a `Tekton user`, I want to use the CLI as a plugin within `tkn` to keep all pipeline-related commands under a single tool.
+
+## Requirements
+
+### Functional Requirements
+
+- Retrieve a `list` of PipelineRuns and TaskRuns from Tekton Results.
+- Describe a `specific` PipelineRun or TaskRun with details.
+- Fetch `logs` of a PipelineRun or TaskRun.
+- Authenticate with the `Tekton Results API` using token-based authentication.
+- Support filtering results based on the flags passed.
+- Allow users to paginate through results interactively, ensuring efficient navigation through large datasets.
+
+### Non-Functional Requirements
+
+- Consistent command syntax, following `tkn` CLI conventions.
+- Formatted output, similar to the tkn CLI.
+- Provide clear and actionable `error messages`.
+- Support partial API response feature
+- Support fetching results in the paginated form using `page_size` and `page_token` 
+
+## Proposal
+
+This proposal introduces a new CLI tool, `tkn-results`, designed to simplify interaction with Tekton Results. The CLI will provide commands to list, describe, and retrieve logs for `PipelineRuns` and `TaskRuns` stored in the Results API. Additionally, it will include authentication support to ensure secure access. The goal is to improve usability and efficiency over the existing CLI while maintaining a familiar command structure for Tekton users. The CLI will also be designed to work as a `plugin` to `tkn`, allowing users to execute commands seamlessly from within their existing Tekton CLI environment.
+
+## Design Details
+
+### Commands
+
+#### Configuration
+
+The configuration will allow users to provide the following options:
+- **Host**: The host address for the client to connect.
+- **Token**: The authentication token for secure API access.
+- **API Path**: The specific API endpoint path.
+- **Skip TLS verification**: Skip TLS verification for the API access (default: false)
+
+These values can be automatically fetched and will have default values when possible.
+
+```sh
+tkn-results config set        # Set authentication and configuration parameters
+tkn-results config reset      # Reset the configuration to default values
+tkn-results config view       # View the current configuration
+```
```

Comment:
I guess this is stored in a file. We need to make sure it's possible for users to have multiple *configuration files* and switch from one to the other using an environment variable (à-la `KUBECONFIG`) or a flag (`--config` ?)

Reply:
There is an option to provide `--kubeconfig` flag which will allow users to use different config file instead of the default one.
Does this make sense or do we want to have environment variable as well?

Reply:
Isn't it the tkn-results configuration more than just the kubeconfig ?

Reply:
No, it is all kubeconfig.
By default, it will fetch from kubeconfig (which will show by default in the `config` command) and the user provided data will also be stored in `Extension` in kubeconfig